scrapypuppeteer/middleware.py

```diff
@@ -203,22 +245,37 @@ def __init__(self,
         self.submit_selectors = submit_selectors
         self.recaptcha_solving = recaptcha_solving
         self._page_responses = dict()
+        self._page_closing = set()
 
     @classmethod
     def from_crawler(cls, crawler: Crawler):
         activation = crawler.settings.get(cls.MIDDLEWARE_ACTIVATION_SETTING, True)
         if not activation:
             raise NotConfigured
         recaptcha_solving = crawler.settings.get(cls.RECAPTCHA_SOLVING_SETTING, True)
+
         try:
             submit_selectors = crawler.settings.getdict(cls.SUBMIT_SELECTORS_SETTING, dict())
         except ValueError:
             submit_selectors = {'': crawler.settings.get(cls.SUBMIT_SELECTORS_SETTING, '')}
+        except Exception as exception:
+            raise ValueError(f"Wrong argument(s) inside {cls.SUBMIT_SELECTORS_SETTING}: {exception}")
+
+        for key in submit_selectors.keys():
+            submit_selector = submit_selectors[key]
+            if isinstance(submit_selector, str):
+                submit_selectors[key] = Click(selector=submit_selector)
+            elif not isinstance(submit_selector, Click):
+                raise ValueError("Submit selector must be str or Click,"
+                                 f"but {type(submit_selector)} provided")
         return cls(recaptcha_solving, submit_selectors)
 
-    @staticmethod
-    def process_request(request, spider):
-        # We don't modify any request, we only work with responses
+    def process_request(self, request, spider):
+        if isinstance(request, PuppeteerRequest):
+            if request.close_page and not request.meta.get('_captcha_submission', False):
+                new_request = request.replace(close_page=False, dont_filter=True)
+                self._page_closing.add(new_request)
+                return new_request
```

Comment:
You can reassign close_page without creating new request, and restore it back when you generate response.